static/scripts/files-storage.js

```diff
@@ -0,0 +1,280 @@
+			},
+			createImageThumbnails: false,
+			method: 'PUT',
+			init() {
+				// this is called on per-file basis
+				this.on('processing', updateUploadProcessingProgress);
+
+				this.on('sending', (file, xhr) => {
+					const { send } = xhr;
+					xhr.send = () => {
+						send.call(xhr, file);
+					};
+				});
+
+				this.on('totaluploadprogress', (_, total, uploaded) => {
+					const realProgress = (uploaded + finishedFilesSize) / ((total + finishedFilesSize) / 100);
+
+					$progress.stop().animate(
+						{ width: `${realProgress}%` },
+						{
+							step(now) {
+								if ($percentage && $percentage.setAttribute) {
+									$percentage.html(`${Math.ceil(now)}%`);
+									$percentage.setAttribute('aria-valuenow', `${Math.ceil(now)}%`);
+								}
+							},
+						},
+					);
+				});
+
+				this.on('queuecomplete', () => {
+					progressBarActive = false;
+					finishedFilesSize = 0;
+
+					$progressBar.fadeOut(50, () => {
+						$form.fadeIn(50);
+						reloadPage('files._file.text.fileAddedSuccess');
+					});
+				});
+
+				this.on('dragover', () => $form.addClass('focus'));
+				this.on('dragleave', () => $form.removeClass('focus'));
+				this.on('dragend', () => $form.removeClass('focus'));
+				this.on('drop', () => $form.removeClass('focus'));
+			},
+		});
+	}
+
+	$('button[data-method="download"]').on('click', (e) => {
+		window.open($(e.currentTarget).attr('data-href'), '_blank');
+		e.stopPropagation();
+	});
+
+	$modals.find('.close, .btn-close').on('click', () => {
+		$modals.modal('hide');
+	});
+
+	$('.btn-file-danger').click((e) => {
+		e.stopPropagation();
+		e.preventDefault();
+		const $dangerModal = $('.danger-modal');
+		$dangerModal.appendTo('body').modal('show');
+	});
+
+	$($renameModal.find('.btn-submit')).click((e) => {
+		e.stopPropagation();
+		e.preventDefault();
+		const fileRecordId = $renameModal.find('#fileRecordId').val();
+		const fileName = $renameModal.find('#newNameInput').val();
+		rename(fileName, fileRecordId);
+		$renameModal.modal('hide');
+	});
+
+	$('#files-storage-component').find('button[data-method="delete"]').on('click', deleteFileClickHandler);
+	$('#files-storage-component').find('a[data-method="delete"]').on('click', deleteFileClickHandler);
+	$('#files-storage-component').find('a[data-method="delete"]').on('keypress', (e) => {
+		if (e.key === 'Enter' || e.key === ' ') {
+			deleteFileClickHandler(e);
+		}
+	});
+
+	$deleteModal.find('.close, .btn-close').on('click', () => {
+		$deleteModal.modal('hide');
+	});
+
+	function populateRenameModal(fileRecordId, oldName, action, title) {
+		const form = $renameModal.find('.modal-form');
+		form.attr('action', action);
+
+		populateModalForm($renameModal, {
+			title,
+			closeLabel: $t('global.button.cancel'),
+			submitLabel: $t('global.button.save'),
+			fields: {
+				fileRecordId,
+				name: oldName,
+			},
+		});
+
+		$renameModal.modal('show');
+	}
+	function fileNameEditClickHandler(e) {
+		e.stopPropagation();
```

Comment:
I google stopPropagation but i have not idea if it is needed on all points or not.
Maybe it make sense. :D